File: backend/models/task.py
from backend.extensions import db
from backend.utils.datetime_utils import utc_now

class Task(db.Model):
    __tablename__ = 'tasks'

    id = db.Column(db.Integer, primary key=True)
    idea_id = db.Column(db.Integer, db.ForeignKey('ideas.id'), nullable=False, index=True)
    title = db.Column(db.String(255), nullable=False)
    description = db.Column(db.Text, nullable=False)
    acceptance_criteria = db.Column(db.Text, nullable=True)
    status = db.Column(db.String(50), nullable=False, default='draft')  # draft, planned, in_progress, done
    is_ai_generated = db.Column(db.Boolean, default=False, nullable=False)
    created_at = db.Column(db.DateTime, default=utc_now)

    # Relationships
    idea = db.relationship('Idea', back_populates='tasks')
    history = db.relationship('TaskHistory', back_populates='task', cascade='all, delete-orphan')

    def __repr__(self):
        return f"<Task {self.id}: {self.title}>"

File: frontend/src/types/task.ts
export type TaskStatus =
    | "draft"
    | "planned"
    | "in_progress"
    | "done";

export interface Task {
    id: number;
    idea_id: number;
    title: string;
    description: string;
    acceptance_criteria: string | null;
    status: TaskStatus;
    is_ai_generated: boolean;
    created_at: string;
}

export interface TaskHistory {
    id: number;
    task_id: number;
    from_status: TaskStatus;
    to_status: TaskStatus;
    changed_at: string;
    note?: string;
}
